app/controllers/omniauth_callbacks_controller.rb

```diff
@@ -0,0 +1,10 @@
+# frozen_string_literal: true
+class OmniauthCallbacksController < Devise::OmniauthCallbacksController
+  def shibboleth
+    Rails.logger.debug "OmniauthCallbacksController#shibboleth: request.env['omniauth.auth']: #{request.env['omniauth.auth']}"
+    # had to create the `from_omniauth(auth_hash)` class method on our User model
+    @user = User.from_omniauth(request.env["omniauth.auth"])
+    set_flash_message :notice, :success, kind: "Shibboleth"
+    sign_in_and_redirect @user
```

Comment:
Can you add a condition here to check if the user is persisted, and only then sign them in? When logging into shib, we expect a user object to already exist in the database, please correct me if this is not a requirement for this app. Refer: https://github.com/emory-libraries/dlp-curate/blob/main/app/controllers/omniauth_callbacks_controller.rb#L7

Reply:
This is a direct copy of the Lux method, which is what I've been told we're mirroring for Blacklight.

Reply:
Correct, but checking if a user exists before signing them in is an improvement which would help us with better error handling. We can skip making the change for now and come back to it if the need arises.

Reply:
No, I'm happy making the changes, it's just that I saw you referring to Curate, and the request was to emulate Lux. Will jump on the changes.

Reply:
Changes have been pushed into the branch.